src/app/components/chefs-table/chefs-table.component.ts

```diff
@@ -33,6 +42,47 @@ export class ChefsTableComponent implements AfterViewInit {
   }
 
   toggleEditMode(row: IChef): void {
-    row.isEditing = !row.isEditing;
+    if (row.isEditing) {
+      this.saveChanges(row);
+      console.log(row)
+      row.isEditing = !row.isEditing;
+
+
+    } else {
+      row.isEditing = !row.isEditing;
```

Comment:
you can remove the else here and put the code
 row.isEditing = !row.isEditing;
 
 out of the else (remove it from the if also)